### ISSUE TITLE
EXC_BAD_ACCESS when allocating memory

### ISSUE THREAD

**Issue by Hovushka** [GPUOpen-LibrariesAndSDKs/VulkanMemoryAllocator#140]
### Library version
2.3.0-vs2017

### Vulkan SDK version
1.2.141.2

### Operating system
MacOS Mojave 10.14.6

### TL;DR
A call to ```vmaCreateImage``` or ```vmaCreateBuffer``` results in a segmentation fault. LLDB shows [this error](https://pastebin.com/EF5gq1dF)

### Verbose
I've decided to try out this library in my Vulkan-hpp project. I've created a [simple wrapper class](https://pastebin.com/vYSQuMNw) around ```VmaAllocator``` for the sake of convenience (```VMA_STATIC_VULKAN_FUNCTIONS ``` and its dynamic counterpart are set to 0).

After that I've noticed the very unstable behaviour of the code. Calling ```create``` functions results in the error described above. Sometimes using ```vmaCreateImage``` and ```vmaCreateBuffer``` outside the class does work, but I couldn't get consistent results.

Other Vulkan programs execute correctly on the hardware.

### I've tried

- Statically linking Vulkan API
- Compiling VMA to a separate static library
- Using some other versions of the library newer than 2.2.0
- Turning on external synchronisation
- Using only C syntax for allocation

### Notes
I couldn't use VMA C++ wrapper since it is outdated and doesn't compile with newer versions of Vulkan. If I am the cause of the issue, then I'm terribly sorry for wasting your time.

**Comment by adam-sawicki-a** [comment]
I'm sorry to hear that you met such severe issues on your platform. Unfortunately I don't have this or any similar platform to test it myself. Your error log indicates that the crash occurs while locking a mutex. You can debug it further to see what's happening inside `VmaBlockVector::Allocate`. In particular, I would suggest:

1. Making sure all the necessary members are properly initialized. For example, in this code:
```cpp
VmaAllocationCreateInfo aci;
aci.usage = flag;
```
Don't you leave all the members of the structure except `usage` uninitialized to garbage data? If the pointer to a custom pool is garbage, that would explain the crash.

(BTW If I am right, I would say this is what happens when you get used to relying on C++ wrappers too much that initialize everything for you 😋 With C structures it doesn't work like this. That's why I always do `VmaAllocationCreateInfo aci = {};` in my sample code - that initializes the structure with zeros.)

2. Checking the way standard C/C++ library is linked with your project, so there are no issues with using its members like `std::mutex`. You can try to use it yourself manually.

**Comment by Hovushka** [comment]
Heh, adding empty curly brackets to the initialisation actually works. I wasn't aware of this weird syntactic С++ behaviour.

I'm extremely grateful for your assistance!
